### PR TITLE
Add detailed stage description for online unsafe recovery

### PR DESCRIPTION
**Pull request by Connor1996**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add detailed stage description for online unsafe recovery

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): a refine on https://github.com/pingcap/docs-cn/pull/9495

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: online-unsafe-recovery.md [hunk 1]
```diff
@@ -128,10 +137,38 @@ PD 下发恢复计划后，会等待 TiKV 上报执行的结果。如上述输
...
+
+```bash
+$ kubectl delete -n ${namespace} pvc ${pvc_name} --wait=false
+```
+
+2. 删除 TiKV Pod ，并等待新创建的 TiKV Pod 加入集群
+
+```bash
+$ kubectl delete -n ${namespace} pod ${pod_name}
+```
```
Comment (Oreoxmt) [inline online-unsafe-recovery.md:174]:
````suggestion
使用 PD Control 执行 [`store remove-tombstone`](/pd-control.md#store-delete--cancel-delete--label--weight--remove-tombstone--limit--storeid---jq"query-string") 命令，从 PD 的元数据中删除已无法恢复的节点。

> **注意：**
>
> - 恢复操作把一些 failed voter 变成了 failed learner，之后还需要 PD 调度经过一些时间将这些 failed learner 移除。只有将 failed learner 都移除后，才能成功进行 `store remove-tombstone`。
> - 建议及时添加新的节点。

从 PD 中移除 Tombstone 状态的 TiKV 后，才能将这些 TiKV 从拓扑中移去。具体操作如下：

<SimpleTab>
<div label="通过 TiUP 部署的节点">

{{< copyable "shell-regular" >}}

```bash
tiup cluster prune <cluster-name>
```

</div>
<div label="通过 TiDB Operator 部署的节点">

1. 删除该 `PersistentVolumeClaim`。

    {{< copyable "shell-regular" >}}

    ```bash
    kubectl delete -n ${namespace} pvc ${pvc_name} --wait=false
    ```

2. 删除 TiKV Pod，并等待新创建的 TiKV Pod 加入集群。

    {{< copyable "shell-regular" >}}

    ```bash
    kubectl delete -n ${namespace} pod ${pod_name}
    ```

</SimpleTab>
````

File: online-unsafe-recovery.md [hunk 2]
```diff
@@ -52,14 +51,24 @@ Online Unsafe Recovery 功能适用于以下场景：
...
 节点移除命令运行成功后，使用 PD Control 执行 [`unsafe remove-failed-stores show`](/pd-control.md#config-show--set-option-value--placement-rules) 命令，查看移除进度。
 
-恢复过程分为多个阶段，每个阶段的进度都以 JSON 格式输出。每一阶段的输出内容包括信息、时间，以及具体的恢复计划。例如：
+恢复过程有多个可能的阶段：
+
+- collect report: 初始阶段，第一次接收 TiKV 的报告获得全局信息。
+- tombstone tiflash learner：删除不健康 Region 中，比其他健康 Peer 要新的 Tiflash learner，防止极端情况。
+- force leader for commit merge：特殊阶段，在有未完成的 commit merge 时出现，优先对有 commit merge 的 Region 进行 force leader，防止极端情况。
+- force leader：强制不健康的 Region 在剩余的健康 Peer 中指定一个成为 Raft leader。
+- demote failed voter：将 Region 的不健康的 Voter 降级为 Learner，如此之后 Region 就能正常的选出 Raft leader。
```
Comment (Oreoxmt) [inline online-unsafe-recovery.md:68]:
```suggestion
- demote failed voter：将 Region 的不健康的 Voter 降级为 Learner，之后 Region 就可以正常地选出 Raft leader。
```

File: online-unsafe-recovery.md [hunk 3]
```diff
@@ -52,14 +51,24 @@ Online Unsafe Recovery 功能适用于以下场景：
...
 
 节点移除命令运行成功后，使用 PD Control 执行 [`unsafe remove-failed-stores show`](/pd-control.md#config-show--set-option-value--placement-rules) 命令，查看移除进度。
 
-恢复过程分为多个阶段，每个阶段的进度都以 JSON 格式输出。每一阶段的输出内容包括信息、时间，以及具体的恢复计划。例如：
+恢复过程有多个可能的阶段：
+
+- collect report: 初始阶段，第一次接收 TiKV 的报告获得全局信息。
+- tombstone tiflash learner：删除不健康 Region 中，比其他健康 Peer 要新的 Tiflash learner，防止极端情况。
+- force leader for commit merge：特殊阶段，在有未完成的 commit merge 时出现，优先对有 commit merge 的 Region 进行 force leader，防止极端情况。
+- force leader：强制不健康的 Region 在剩余的健康 Peer 中指定一个成为 Raft leader。
```
Comment (Oreoxmt) [inline online-unsafe-recovery.md:67]:
```suggestion
- `tombstone tiflash learner`：删除不健康的 Region 中，比其他健康 Peer 要新的 TiFlash learner，防止极端情况。
- `force leader for commit merge`：特殊阶段，在有未完成的 commit merge 时出现，优先对有 commit merge 的 Region 进行 force leader，防止极端情况。
- `force leader`：强制不健康的 Region 在剩余的健康 Peer 中指定一个成为 Raft leader。
```

File: online-unsafe-recovery.md [hunk 4]
```diff
@@ -52,14 +51,24 @@ Online Unsafe Recovery 功能适用于以下场景：
...
 
-恢复过程分为多个阶段，每个阶段的进度都以 JSON 格式输出。每一阶段的输出内容包括信息、时间，以及具体的恢复计划。例如：
+恢复过程有多个可能的阶段：
+
+- `collect report`：初始阶段，第一次接收 TiKV 的报告获得的全局信息。
+- `tombstone tiflash learner`：删除不健康的 Region 中，比其他健康 Peer 要新的 TiFlash learner，防止极端情况。
+- `force leader for commit merge`：特殊阶段，在有未完成的 commit merge 时出现，优先对有 commit merge 的 Region 进行 force leader，防止极端情况。
+- `force leader`：强制不健康的 Region 在剩余的健康 Peer 中指定一个成为 Raft leader。
+- `demote failed voter`：将 Region 不健康的 Voter 降级为 Learner，之后 Region 就可以正常地选出 Raft leader。
+- create empty region：创建空 Region 补足 key range 空洞，主要针对的是某些 Region 的所有副本所在的 Store 都损坏了。
```
Comment (Oreoxmt) [inline online-unsafe-recovery.md:69]:
```suggestion
- `create empty region`：创建一个空 Region 来填充 key range 的空缺，主要针对的是某些 Region 的所有副本所在的 Store 都损坏了。
```

File: online-unsafe-recovery.md [hunk 5]
```diff
@@ -52,14 +51,24 @@ Online Unsafe Recovery 功能适用于以下场景：
...
 ### 第 2 步：查看进度等待结束
 
 节点移除命令运行成功后，使用 PD Control 执行 [`unsafe remove-failed-stores show`](/pd-control.md#config-show--set-option-value--placement-rules) 命令，查看移除进度。
 
-恢复过程分为多个阶段，每个阶段的进度都以 JSON 格式输出。每一阶段的输出内容包括信息、时间，以及具体的恢复计划。例如：
+恢复过程有多个可能的阶段：
+
+- `collect report`：初始阶段，第一次接收 TiKV 的报告获得的全局信息。
+- `tombstone tiflash learner`：删除不健康的 Region 中，比其他健康 Peer 要新的 TiFlash learner，防止极端情况。
+- `force leader for commit merge`：特殊阶段，在有未完成的 commit merge 时出现，优先对有 commit merge 的 Region 进行 force leader，防止极端情况。
```
Comment (Oreoxmt) [inline online-unsafe-recovery.md:66]:
```suggestion
- `force leader for commit merge`：特殊阶段。在有未完成的 commit merge 时出现，优先对有 commit merge 的 Region 进行 `force leader`，防止极端情况。
```

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Review by v01dstar — APPROVED** [review]

File: online-unsafe-recovery.md [hunk 6]
```diff
@@ -128,10 +137,51 @@ PD 下发恢复计划后，会等待 TiKV 上报执行的结果。如上述输
...
+
+2. 删除 TiKV Pod，并等待新创建的 TiKV Pod 加入集群。
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    kubectl delete -n ${namespace} pod ${pod_name}
+    ```
+
+</SimpleTab>
```
Comment (Oreoxmt) [inline online-unsafe-recovery.md:187]:
```suggestion
</div>
</SimpleTab>
```

File: online-unsafe-recovery.md [hunk 7]
```diff
@@ -128,10 +137,52 @@ PD 下发恢复计划后，会等待 TiKV 上报执行的结果。如上述输
...
-当恢复完成后，你可以尝试执行一些简单 SQL 查询或写入操作确保数据可以读写。执行完成后，可能会导致数据索引不一致，请使用 `ADMIN CHECK`、`ADMIN RECOVER`、`ADMIN CLEANUP` 命令进行数据索引的一致性检查及恢复。
+执行完成后，可能会导致数据索引不一致。请使用 SQL 的 `ADMIN CHECK`、`ADMIN RECOVER`、`ADMIN CLEANUP` 命令对受影响的表（从 `"Unsafe recovery finished"` 输出的 `"Affected table ids"` 可知）进行数据索引的一致性检查及恢复。
 
 > **注意：**
 >
 > 数据可以读写并不代表没有数据丢失。
+
+### 第 4 步：移除无法恢复的节点（可选）
+
+使用 PD Control 执行 [`store remove-tombstone`](/pd-control.md#store-delete--cancel-delete--label--weight--remove-tombstone--limit--storeid---jq"query-string") 命令，从 PD 的元数据中删除已无法恢复的节点。
```
Comment (Oreoxmt) [inline online-unsafe-recovery.md:150]:
```suggestion
使用 PD Control 执行 [`store remove-tombstone`](/pd-control.md#store-delete--cancel-delete--label--weight--remove-tombstone--limit--store_id---jqquery-string) 命令，从 PD 的元数据中删除已无法恢复的节点。
```

File: online-unsafe-recovery.md [hunk 8]
```diff
@@ -52,14 +51,24 @@ Online Unsafe Recovery 功能适用于以下场景：
...
 
 ### 第 2 步：查看进度等待结束
 
 节点移除命令运行成功后，使用 PD Control 执行 [`unsafe remove-failed-stores show`](/pd-control.md#config-show--set-option-value--placement-rules) 命令，查看移除进度。
 
-恢复过程分为多个阶段，每个阶段的进度都以 JSON 格式输出。每一阶段的输出内容包括信息、时间，以及具体的恢复计划。例如：
+恢复过程有多个可能的阶段：
+
+- `collect report`：初始阶段，第一次接收 TiKV 的报告获得的全局信息。
+- `tombstone tiflash learner`：删除不健康的 Region 中，比其他健康 Peer 要新的 TiFlash learner，防止极端情况。
```
Comment (TomShawn) [inline online-unsafe-recovery.md:65]: 不知是不是这个意思？
```suggestion
- `tombstone tiflash learner`：在不健康的 Region 中，删除比其他健康 Peer 要新的 TiFlash learner，防止极端情况。
```

File: online-unsafe-recovery.md [hunk 9]
```diff
@@ -52,14 +51,24 @@ Online Unsafe Recovery 功能适用于以下场景：
...
+- `force leader for commit merge`：特殊阶段。在有未完成的 commit merge 时出现，优先对有 commit merge 的 Region 进行 `force leader`，防止极端情况。
+- `force leader`：强制不健康的 Region 在剩余的健康 Peer 中指定一个成为 Raft leader。
```
Comment (TomShawn) [inline online-unsafe-recovery.md:67]: 这里的「指定」是「选举」吗？

Comment (Connor1996) [reply online-unsafe-recovery.md:67]: 这个跳过了选举，直接指定 leader

File: online-unsafe-recovery.md (hunk 8 above)
Comment (Connor1996) [reply online-unsafe-recovery.md:65]: 是的；比其他健康 Peer 要新的 TiFlash learner 就是极端情况，可能会造成 panic

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**